Z/ZlibNG/build_tarballs.jl

```diff
@@ -0,0 +1,61 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "ZlibNG"
+version = v"1.2.12"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/zlib-ng/zlib-ng.git", "0c118a14dd56d0484a8ca432cf801b9a6e6d024a")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/zlib-ng
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=$prefix \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_C_FLAGS="-O3" \
+    -DZLIB_COMPAT=ON \
+    ..
+make -j${nproc}
+make install
+
+if [[ "${target}" == *-mingw* ]]; then
+    cp "${WORKSPACE}/srcdir/zlib-ng/build/libzlib1.dll" "${libdir}/libz.dll"
+fi
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("i686", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("x86_64", "macos"; ),
+    Platform("aarch64", "macos"; ),
+    Platform("i686", "windows"; ),
+    Platform("x86_64", "windows"; )
+]
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libz", :libz)
```

Comment:
This can't have the same soname as zlib https://github.com/JuliaPackaging/Yggdrasil/blob/1e926895c26b578d145db8d483654a91289bac4c/Z/Zlib/build_tarballs.jl#L40 otherwise this will never be loaded.

Reply:
Is `LibraryProduct("libz", :libzng)` better?

Reply:
Not really, it's the fact that the build produces a file with [soname](https://en.wikipedia.org/wiki/Soname) the problem.  The second argument to [`LibraryProduct`](https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.LibraryProduct) is just the name of the Julia variable used in the generated JLL package, it isn't very important

Reply:
Gotcha. Yeah so I think we won't go the `ZLIB_COMPAT` route, then there won't be a clash in sonames.